pkg/cfg/config.go

```diff
@@ -531,8 +579,7 @@ func (c *config) resolveEnvKey(prefix string, key string) string {
 		key = strings.Join([]string{prefix, key}, ".")
 	}
 
-	rp := regexp.MustCompile(`\[(\d)\]`)
-	matches := rp.FindAllStringSubmatch(key, -1)
+	matches := keyToEnvRegexp.FindAllStringSubmatch(key, -1)
```

Comment:
So we are transforming foo[3] to foo.3 here (or in the loop following this)? What happens if we get foo[32]? That doesn't match with a single digit, right?

Reply:
added a `+`